config/config.go

```diff
@@ -27,14 +35,24 @@ func ParseAsV2_0_0(data []byte) (types.Config, error) {
 		return types.Config{}, err
 	}
 
-	var keyMap map[interface{}]interface{}
-	if err := yaml.Unmarshal(data, &keyMap); err != nil {
-		return types.Config{}, err
-	}
+	var r report.Report
 
-	if err := assertKeysValid(keyMap, reflect.TypeOf(Config{})); err != nil {
-		return types.Config{}, err
+	nodes := yaml.UnmarshalToNode(data)
+	if nodes == nil {
+		fmt.Println("Failed to unmarshal yaml for reporting line numbers.")
```

Comment:
I'm not a fan of these prints. This package is consumed by other applications. They should be able to print the way they like. Can we just return the report instead?